src/components/shared/modals/modalWithInput/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React, {useEffect} from 'react';
+import {Button, Modal, Text} from 'native-base';
+import Input from 'components/shared/input';
+import styles from './styles';
+import {useForm} from 'react-hook-form';
+import {schema} from './validations';
+import {yupResolver} from '@hookform/resolvers/yup';
+import {Keyboard} from 'react-native';
+
+interface Props {
+  buttonText: string;
+  headerText: string;
+  firstInputName: string;
+  secondInputName?: string;
+  onSubmit: () => void;
```

Comment:
probably, we might want to make use of the data entered on the form with this onSubmit function. Let's update the typing for this purpose
```suggestion
  onSubmit: (data: {[x: string]: string}) => void;
```

Reply:
you are right! forgot to update this interface, done.